dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/MeController.java

```diff
@@ -457,13 +456,12 @@ private void merge( User currentUser, User user )
         currentUser.setInterests( stringWithDefault( user.getInterests(), currentUser.getInterests() ) );
         currentUser.setLanguages( stringWithDefault( user.getLanguages(), currentUser.getLanguages() ) );
 
-        // Reset user secret when re-enable twoFA
-        if ( !currentUser.isTwoFA() && user.isTwoFA() )
+        // TODO: NOT ALLOWED AFTER 13332, breaks current API/UI (2.39) 2.38
+        // backport later
+        if ( currentUser.getTwoFA() != user.getTwoFA() )
         {
-            currentUser.setSecret( Base32.random() );
+            throw new IllegalArgumentException( "Not allowed to disable 2FA with this endpoint!" );
```

Comment:
Should we use a predefined `ErrorCode` here instead of free text, and ensure we catch this exception in `CrudControllerAdvice`?